File: languageSchoolProject/src/main/java/com/langschool/languageschoolproject/service/TeacherService.java
package com.langschool.languageschoolproject.service;

import com.langschool.languageschoolproject.entity.Teacher;
import com.langschool.languageschoolproject.repo.TeacherRepo;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class TeacherService {

    @Autowired
    private TeacherRepo repository;
    public Teacher saveTeacher(Teacher teacher)
    {
        return repository.save(teacher);
    }

    public List<Teacher> saveTeachers(List<Teacher> teachers)
    {
        return repository.saveAll(teachers);
    }

    public List<Teacher> getTeachers()
    {
        return repository.findAll();
    }

    public Teacher getTeacherById(int id)
    {
        return repository.findById(id).orElse(null);
    }
    public Teacher getTeacherBySubject (String subj)
    {
        return repository.findBySubject(subj);

    }

    public String deleteTeacher(int id)
    {
        repository.deleteById(id);
        return "Teacher with id "+id+" removed!";
    }

    public Teacher updateTeacher(Teacher teacher)
    {
        Teacher existingTeacher=repository.findById(teacher.getId()).orElse(null);
        existingTeacher.setName(teacher.getName());
        existingTeacher.setLastName(teacher.getLastName());
        existingTeacher.setSubject(teacher.getSubject());
        existingTeacher.setLogin(teacher.getLogin());
        existingTeacher.setPassword(teacher.getPassword());
        return repository.save(existingTeacher);
    }

}

File: react/my-app/src/services/TeacherService.js
import axios from 'axios';

const TEACHER_API_BASE_URL = "http://localhost:9192/api";

class TeacherService {

    getTeachers(){
        return axios.get(TEACHER_API_BASE_URL + "/teachers");
    }

    createTeacher(teacher){
        return axios.post(TEACHER_API_BASE_URL + "/addteacher" , teacher);
    }

    getTeacherById(teacherId){
        return axios.get(TEACHER_API_BASE_URL + '/teacherById/' + teacherId);
    }

    updateTeacher(teacher){
        return axios.put(TEACHER_API_BASE_URL + '/updateTeacher/', teacher);
    }

    deleteTeacher(teacherId){
        return axios.delete(TEACHER_API_BASE_URL + '/deleteTeacher/' + teacherId);
    }
}

export default new TeacherService()